src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -80,18 +81,80 @@ class ActivityFeed extends React.Component<Props, State> {
     approvalCommentFormSubmitHandler = (): void => this.setState({ isInputOpen: false });
 
     /**
-     * Creates a comment
+     * Add a placeholder pending feed item.
      *
-     * @param {string} text - Comment text
-     * @param {boolean} hasMention - If this comment contains an at mention
+     * @param {Object} itemBase - Base properties for item to be added to the feed as pending.
+     * @return {void}
+     */
+    addPendingItem = (itemBase: Object): void => {
+        const { currentUser } = this.props;
+        // create a placeholder pending feed item
+        const date = new Date().toISOString();
+        const feedItem = {
+            created_at: date,
+            created_by: currentUser,
+            modified_at: date,
+            isPending: true,
+            ...itemBase
+        };
+
+        const { feedItems } = this.state;
+        feedItems.unshift(feedItem);
+        this.setState({ feedItems });
+    };
+
+    /**
+     * Replace a pending feed item with actual feed item data.
+     *
+     * @param {Comment | Task} feedItem - API returned feed item data.
+     * @param {string} uuid - Unique ID of the pending item to replace.
+     * @return {void}
+     */
+    updatePendingItem = (feedItem: Comment | Task, uuid: string): void => {
+        let itemIndex = null;
+        const { feedItems } = this.state;
+        feedItems.find((item, index) => {
+            if (item.id === uuid) {
+                itemIndex = index;
+                return true;
+            }
+            return false;
+        });
+
+        // Replace item in the feed items or set as most recent item.
+        if (itemIndex !== null) {
+            feedItems[itemIndex] = feedItem;
+        } else {
+            feedItems.unshift(feedItem);
+        }
+
+        this.setState({ feedItems });
+    };
+
+    /**
+     * Create a comment, and make a pending item to be replaced once the API is successful.
+     *
+     * @param {any} args - Data returned by the Comment component on comment creation.
      * @return {void}
      */
     createComment = ({ text, hasMention }: { text: string, hasMention: boolean }): void => {
-        // create a placeholder pending comment
-        // create actual comment and send to Box V2 api
-        // call user passed in handlers.comments.create, if it exists
-        const createComment = getProp(this.props, 'handlers.comments.create', noop);
-        createComment(text, hasMention);
+        const uuid = uniqueId();
```

Comment:
uniqueId('comment_');

so that its prefixed specific to things used in this file

Reply:
Done!